### ISSUE TITLE
oemof does not work with pyomo 5.4.x at the moment

### ISSUE THREAD

**Issue by uvchik** [oemof/oemof#448]
I do not know if it is a pyomo bug or an API change but we should find out.

**Comment by ckaldemeyer** [comment]
https://github.com/oemof/oemof/issues/444

**Comment by ckaldemeyer** [comment]
This is annoying since I need version 5.4 for some additional stuff..

**Comment by ckaldemeyer** [comment]
I have tried to locate the problem and from my opinion it seems to be related to changes within the Pyomo API.

See the following thread for my question on the Pyomo mailing list: https://groups.google.com/forum/#!topic/pyomo-forum/MQbSL2BOOG0

**Comment by ckaldemeyer** [comment]
The quick release history for the latest release underpins our assumption: https://github.com/Pyomo/pyomo/releases

Let's wait for their answer ;-)

**Comment by ckaldemeyer** [comment]
> The quick release history for the latest release underpins our assumption: https://github.com/Pyomo/pyomo/releases
> 
> Let's wait for their answer ;-)

Here's the answer:

```
Cord:

There haven't been API changes, and the expression system didn't change.  This looks like an issue with the index value used to define the constraint group.

There were changes in the logic for testing such values.  

--Bill
```

**Comment by ckaldemeyer** [comment]
This shows how to fix it so we can think about fixing it and then switch to Pyomo >= 5.4.

**Comment by uvchik** [comment]
I just talked to @ckaldemeyer and he will try to fix it this week. I think we should wait with the release so that the new release works with the actual oemof version.

**Comment by jnnr** [comment]
This is a real bug, so I am o.k. with postponing the release a bit to include the fix.

**Comment by simnh** [comment]
I will try to fix this today...

**Comment by simnh** [comment]
Please check out my pull request #449

**Comment by uvchik** [comment]
Closed in favour of #449.
